### ISSUE TITLE
Use indentation to define continuation instead of literal paragraph

### ISSUE THREAD

**Issue by DavidGamba** [asciidoctor/asciidoctor#623]
One thing I don't like/agree with about AsciiDoc syntax is that it uses
indentation to define literal paragraphs.

I would like for you to change that behaviour so instead of using it like that
it is used for list continuation:

Example:

```
= Title here

* a

* b

  ** b a

    This should NOT be a literal paragraph but it is unfortunately

  ** b b

    ----
    This SHOULD be an optional way of defining a literal paragraph inside a list.
    ----

  ** b c
+
----
Having to define the literal paragraph at the beggining is acceptable
----

  ** b d
+
But this is not readable, without the indentation you lose the list flow.

Text outside the list is at the same level as text inside the list.
```

I know this change would break compatibility but having indentation is far more
readable than not.

If the purpose of AsciiDoc is to write html or docfiles in an easier format than
those formats then it has acomplished that goal, however, if one of the goals is
to be able to read the file source itself as a readable file then it is missing
the indentation part.

**Comment by 997293** [comment]
Dan,

Do you think it's possible to change the fundamental philosophy of asciidoc so that asciidoc can adopt the list continuation syntax of markdown?

The way markdown deals with list continuation is just so beautiful, and the way asciidoc deals with list continuation is just so "unreadable", that I believe if asciidoc can adopt the markdown list continuation syntax, it would be the best lightweight markup language ever existed.

**Comment by mojavelinux** [comment]
@997293 I have exactly the opposite opinion as you. To me, the strategy Markdown uses is extremely tedious and finicky. I believe the compromise I suggested is the right path forward.

**Comment by Mogztter** [comment]
What about ["Attaching to an ancestor list"](https://asciidoctor.org/docs/user-manual/#complex-list-content) under [this proposal](https://github.com/asciidoctor/asciidoctor/issues/623#issuecomment-24145829)?

> Each blank line that precedes the list continuation signals a move up one level of nesting...

I have mixed feeling about this syntax under this proposal. In the following example, using Lorem Ipsum instead of explicit text, can you tell where the list ends?

```
* 1

ad litora torquent per conubia nostra.

* 2

elit et tincidunt rutrum.

** 2.1

consectetur adipiscing elit.

*** 2.1.1

quis dapibus quam.

nec scelerisque turpis. +
torquent per conubia nostra.

+
risus et aliquet dictum.


+
ante ipsum primis in faucibus


faucibus fermentum.
+
suscipit metus lectus.


vehicula augue lectus non nisl.
```

What about a symbol to control the depth?


```
* 1

ad litora torquent per conubia nostra.

* 2

elit et tincidunt rutrum.

** 2.1

consectetur adipiscing elit.

*** 2.1.1

quis dapibus quam.

nec scelerisque turpis. +
torquent per conubia nostra.

<
risus et aliquet dictum.

<<
ante ipsum primis in faucibus


faucibus fermentum.
+
suscipit metus lectus.


vehicula augue lectus non nisl.
```

We could even use this symbol to move out of the list. For instance, in the above example, using `<<<` will effectively put the content outside of the list because we have a depth of 3.
Here's a concret example:

```
* 1
* 1.1
* 1.1.1
* 1.1.1.1

<<<<
I want to break free
```

Having said that, I think we can keep the double blank lines in this case because it's not used anywhere else in the list so it's much easier to spot where the list ends (i.e. as soon as you see a double blank lines, the list ends).
I just don't think that we should introduce the double blank lines if we already use it to attach an item to an ancestor list (parent of parent).

It might be a bit off-topic (sorry :sweat_smile:) but I think it's important to consider how this proposal will fit with other elements of the AsciiDoc syntax.

**Comment by mojavelinux** [comment]
> I just don't think that we should introduce the double blank lines if we already use it to attach an item to an ancestor list (parent of parent).

I don't like the double blank lines even for how they are used today (to attach to a parent list item). That's one of the weaker parts of the AsciiDoc syntax and quite fragile. Though I believe that's really a separate issue from how to attach to the current level.

**Comment by mojavelinux** [comment]
For the record, my proposal can be found in https://github.com/asciidoctor/asciidoctor/issues/686 rather than this thread. We seem to have two parallel discussions going (albeit with a slightly different premise).

**Comment by mojavelinux** [comment]
The original proposal in this thread would unnecessarily break countless documents, so I don't even think it could be considered.

**Comment by Mogztter** [comment]
> For the record, my proposal can be found in #686 rather than this thread. We seem to have two parallel discussions going (albeit with a slightly different premise).

Indeed... though the goal is the same: make complex lists more readable.
In my opinion, complex lists are hard to parse because we have to use the `+` continuation symbol. It makes the content get kind of cluttered.

I agree that indentation can improve readability but not as much as removing the explicit `+` continuation symbol:


```
The following types of animals exist:

* Birds
* Mammals
** Land-Mammals

Here's how to describe a land-mammal of these in java:

[source,java]
----
String s = "Gorilla";
----

Subtypes:

*** Apes

WARNING: humans are apes.

*** Rodents

** Sea-Mammals

NOTE: Sea-mammals breath air, not water

* Insects
```

In the example above, the readability is good and the markup is minimal.
The syntax is also very familiar as it seems that we are using section titles:

```
The following types of animals exist:

== Birds
== Mammals
=== Land-Mammals

Here's how to describe a land-mammal of these in java:

[source,java]
----
String s = "Gorilla";
----

Subtypes:

==== Apes

WARNING: humans are apes.

==== Rodents

=== Sea-Mammals

NOTE: Sea-mammals breath air, not water

== Insects
```

The main drawback is that we now have to "close" the list.
The "compact" list proposal might be a good solution to mitigate this issue.

We could also use an open block to determine the boundaries of complex lists or use my proposal move out of the list using the `^` symbol (replacing 5 continuation list symbols `+` with a single "closing" symbol `^`).

If we want a smooth transition, we could use an option on the list to enable this feature.

**Comment by timjs** [comment]
If the main concern is to keep AsciiDoc blok based and keep blocks at the left margin, why not use a block delimiter to sign for a "complex list", or just use an open block delimiter for this purpose?

```
* simple
* list

--
* Complex 

list with 

----
Code 
----

** And multiple

levels
```

**Comment by mojavelinux** [comment]
> If the main concern is to keep AsciiDoc blok based and keep blocks at the left margin, why not use a block delimiter to sign for a "complex list"

A proposal we will be considering in the AsciiDoc language project is to allow the list continuation to be omitted if the principal text is directly adjacent to an open block (and possibly any block). For example:

```
* simple list item
* complex list item
~~~~
paragraph

----
code
----

paragraph
~~~~
```

which is a simplification of this example, which is already supported:

```
* simple list item
* complex list item
+
--
paragraph

----
code
----

paragraph
--
``

If that proposal gets accepted, it will be implemented in Asciidoctor.

**Comment by DavidGamba** [comment]
@mojavelinux Where is that proposal found? I tried here but I can't quite find it.

https://gitlab.eclipse.org/eclipse/asciidoc/asciidoc-lang
https://projects.eclipse.org/projects/technology.asciidoc

**Comment by mojavelinux** [comment]
I said "a proposal we will be considering" (meaning there isn't an issue filed for it yet). It came up on the asciidoc-lang-dev mailing list, and I had previously mentioned it somewhere in this issue tracker. (Right now, I can't find the discussion, but I know they happened). Once the formal proposal has been submitted, I'll link to it.
